requests/adapters.py

```diff
@@ -418,11 +421,14 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                        "timeout tuple, or a single float to set "
                        "both timeouts to the same value".format(timeout))
                 raise ValueError(err)
-        else:
+        elif timeout is not None:
             timeout = TimeoutSauce(connect=timeout, read=timeout)
```

Comment:
Nope, they aren't the same. They both validate to the identity, but that identity is then used as the actual timeout. Note: https://github.com/shazow/urllib3/blob/master/urllib3/util/timeout.py#L33